### PR TITLE
tools: add the Binlog Slave Client user guide

### PR DESCRIPTION
**Pull request by CaitinChen**
Via https://github.com/pingcap/docs-cn/pull/938
@lilin90 @WangXiangUSTC PTAL

### CODE REVIEW

File: tools/binlog-slave-client.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+title: Binlog Slave Client User Guide
+category: tools
+---
+
+# Binlog Slave Client User Guide
+
+Binlog Slave Client is used to parse the binlog data and output the data in a specific format to Kafka. This feature is introduced because currently, Drainer supports outputting data in a format including MySQL, TiDB, TheFlash, and pb, which does not meet users' requirement for outputting data to, for example, Elasticsearch and Hive. After data is output to Kafka, the user writes code to read data from Kafka and then processes teh data.
```
Comment (WangXiangUSTC) [inline tools/binlog-slave-client.md]: typo teh?

**Comment by CaitinChen** [comment]
@WangXiangUSTC Thanks. @lilin90 PTAL

File: tools/binlog-slave-client.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+---
+title: Binlog Slave Client User Guide
```
Comment (lilin90) [inline tools/binlog-slave-client.md]: Add a summary line after title.

File: tools/binlog-slave-client.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+category: tools
+---
+
+# Binlog Slave Client User Guide
+
+Binlog Slave Client is used to parse the binlog data and output the data in a specific format to Kafka. This feature is introduced because currently, Drainer supports outputting data in a format including MySQL, TiDB, TheFlash, and pb, which does not meet users' requirement for outputting data to, for example, Elasticsearch and Hive. After data is output to Kafka, the user writes code to read data from Kafka and then processes the data.
```
Comment (lilin90) [inline tools/binlog-slave-client.md]:
```suggestion
Binlog Slave Client is used to parse the binlog data and output the data in a specific format to Kafka. Currently, Drainer supports outputting data in multiple formats including MySQL, TiDB, TheFlash, and pb. But sometimes users have customized requirements for outputting data to other formats, for example, Elasticsearch and Hive, so this feature is introduced. After data is output to Kafka, the user writes code to read data from Kafka and then processes the data.
```

File: tools/binlog-slave-client.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+* `ClusterID`: the cluster ID of the TiDB cluster
+
+You can use Driver by quoting the Driver code in package and refer to the example code provided by Driver to learn how to use Driver and parse the binlog data. 
+
+Currently, there are two examples:
+
+* Using Driver to synchronize data to MySQL. This example shows how to convert a binlog to SQL
+* Using Driver to print data
+
+The Driver address: [Binlog Slave Driver](https://github.com/pingcap/tidb-tools/tree/master/tidb_binlog/driver)。
```
Comment (lilin90) [inline tools/binlog-slave-client.md]: Moving this address to the first sentence in this section is more user-friendly.

File: tools/binlog-slave-client.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  optional DMLData dml_data = 3;
+  optional DDLData ddl_data = 4;
+}
+```
+
+For the definition of the data format, see [`binlog.proto`](https://github.com/pingcap/tidb-tools/blob/master/tidb_binlog/slave_binlog_proto/proto/binlog.proto).
+
+### Driver
+
+TiDB-Tools provides Driver which is used to read the binlog data in Kafka. It has the following features:
```
Comment (lilin90) [inline tools/binlog-slave-client.md]:
```suggestion
The [TiDB-Tools](https://github.com/pingcap/tidb-tools/) project provides [Driver](https://github.com/pingcap/tidb-tools/tree/master/tidb_binlog/driver), which is used to read the binlog data in Kafka. It has the following features:
```

File: tools/binlog-slave-client.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+You need to configure the following information when using Driver:
+
+* `KafkaAddr`: the address of the Kafka cluster
+* `CommitTS`: from which `commit ts` to start reading the binlog
+* `Offset`: from which Kafka `offset` to start reading data. If `CommitTS` is set, you needn't configure this parameter
+* `ClusterID`: the cluster ID of the TiDB cluster
+
+You can use Driver by quoting the Driver code in package and refer to the example code provided by Driver to learn how to use Driver and parse the binlog data. 
+
+Currently, there are two examples:
```
Comment (lilin90) [inline tools/binlog-slave-client.md]: Generally, limit the use of the "there be" pattern.

-> Currently, two examples are provided:

File: tools/binlog-slave-client.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+* Using Driver to synchronize data to MySQL. This example shows how to convert a binlog to SQL
+* Using Driver to print data
+
+The Driver address: [Binlog Slave Driver](https://github.com/pingcap/tidb-tools/tree/master/tidb_binlog/driver)。
+
+> **Notes:**
+>
+> - The example code only shows how to use Driver. If you want to use Driver in the production environment, you need to optimize the code.
+> - Currently, only Driver for golang and the example code are supported. If you want to use other languages, you need to generate the code file in the corresponding language based on the binlog proto file and develop an application to read the binlog data in Kafka, parse the data, and output the data to the downstream. You are also welcome to optimize the example code and submit the example code of other languages to [TiDB-Tools](https://github.com/pingcap/tidb-tools).
```
Comment (lilin90) [inline tools/binlog-slave-client.md]: The first sentence here is a little bit inappropriate. We can put it this way:

-> Currently, **only the Golang version of Driver and example code are available.**

**Comment by CaitinChen** [comment]
@lilin90 Thanks. PTAL again.

File: tools/binlog-slave-client.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
+---
+title: Binlog Slave Client User Guide
+summary: Learn how to use Binlog Slave Client.
```
Comment (lilin90) [inline tools/binlog-slave-client.md]: The information conveyed in this sentence is a little bit too less. The summary sentence is intended to benefit SEO. We need to think how to convey useful info concisely. Perhaps we can change it to:

Use Binglog Slave Client to parse the binlog data and output the data in a specific format to Kafka.

**Comment by CaitinChen** [comment]
@lilin90 Thanks a lot. PTAL again.

**Pull request merged**